x-pack/plugins/ml/public/data_frame/pages/data_frame_new_pivot/wizard.tsx

```diff
@@ -0,0 +1,181 @@
+ */
+
+import React, { Fragment, SFC, useContext, useRef, useState } from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import { EuiSteps, EuiStepStatus } from '@elastic/eui';
+
+import { WizardNav } from '../../components/wizard_nav';
+
+import {
+  DefinePivotExposedState,
+  DefinePivotForm,
+  DefinePivotSummary,
+  getDefaultPivotState,
+} from '../../components/define_pivot';
+
+import {
+  getDefaultJobCreateState,
+  JobCreateForm,
+  JobCreateSummary,
+} from '../../components/job_create';
+
+import { getDataFrameRequest } from '../../common';
+import {
+  getDefaultJobDetailsState,
+  JobDetailsForm,
+  JobDetailsSummary,
+} from '../../components/job_details';
+
+import { IndexPatternContext } from '../../common';
+
+enum WIZARD_STEPS {
+  DEFINE_PIVOT,
+  JOB_DETAILS,
+  JOB_CREATE,
+}
+
+interface DefinePivotStepProps {
+  isCurrentStep: boolean;
+  pivotState: DefinePivotExposedState;
+  setCurrentStep: React.Dispatch<React.SetStateAction<WIZARD_STEPS>>;
+  setPivot: React.Dispatch<React.SetStateAction<DefinePivotExposedState>>;
+}
+
+const DefinePivotStep: SFC<DefinePivotStepProps> = ({
+  isCurrentStep,
+  pivotState,
+  setCurrentStep,
+  setPivot,
+}) => {
+  const definePivotRef = useRef(null);
+
+  return (
+    <Fragment>
+      <div ref={definePivotRef} />
+      {isCurrentStep && (
+        <Fragment>
+          <DefinePivotForm onChange={setPivot} overrides={pivotState} />
+          <WizardNav
+            next={() => setCurrentStep(WIZARD_STEPS.JOB_DETAILS)}
+            nextActive={pivotState.valid}
+          />
+        </Fragment>
+      )}
+      {!isCurrentStep && <DefinePivotSummary {...pivotState} />}
+    </Fragment>
+  );
+};
+
+export const Wizard: SFC = React.memo(() => {
+  // indexPattern from context
+  const indexPattern = useContext(IndexPatternContext);
+
+  if (indexPattern === null) {
+    return null;
+  }
+
+  // The current WIZARD_STEP
+  const [currentStep, setCurrentStep] = useState(WIZARD_STEPS.DEFINE_PIVOT);
+
+  // The DEFINE_PIVOT state
+  const [pivotState, setPivot] = useState(getDefaultPivotState());
+
+  // The JOB_DETAILS state
+  const [jobDetailsState, setJobDetails] = useState(getDefaultJobDetailsState());
+
+  const jobDetails =
+    currentStep === WIZARD_STEPS.JOB_DETAILS ? (
+      <JobDetailsForm onChange={setJobDetails} overrides={jobDetailsState} />
+    ) : (
+      <JobDetailsSummary {...jobDetailsState} />
+    );
+
+  // The JOB_CREATE state
+  const [jobCreateState, setJobCreate] = useState(getDefaultJobCreateState);
+
+  const jobCreate =
+    currentStep === WIZARD_STEPS.JOB_CREATE ? (
+      <JobCreateForm
+        jobId={jobDetailsState.jobId}
+        jobConfig={getDataFrameRequest(indexPattern.title, pivotState, jobDetailsState)}
+        onChange={setJobCreate}
+        overrides={jobCreateState}
+      />
+    ) : (
+      <JobCreateSummary />
+    );
+
+  // scroll to the currently selected wizard step
```

Comment:
Is this commented out section still needed?

Reply:
I want to re-enable this but it's broken for now because of a previous refactor. I added a note to the Github project to address it later.